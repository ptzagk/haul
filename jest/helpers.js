/**
 * Copyright 2017-present, Callstack.
 * All rights reserved.
 *
 * @flow
 */

'use strict';

function flushPromises() {
  return new Promise(resolve => setTimeout(resolve));
}

module.exports = {
  flushPromises,
};
