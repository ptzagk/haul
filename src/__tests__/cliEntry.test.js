/**
 * Copyright 2017-present, Callstack.
 * All rights reserved.
 *
 * @flow
 */

import stripAnsi from 'strip-ansi';
import { flushPromises } from 'jest/helpers';
import run from '../cliEntry';
import init from '../commands/init';

jest.mock('../commands/init', () => {
  return {
    name: 'init',
    description: 'Generates necessary configuration files',
    action: jest.fn(),
  };
});

const originalConsoleLog = console.log;

beforeEach(() => {
  // $FlowFixMe
  console.log = jest.fn();
});

afterEach(() => {
  // $FlowFixMe
  console.log = originalConsoleLog;
  jest.clearAllMocks();
});

test('display help', () => {
  run(['help']);
  run(['--help']);
  const { calls } = console.log.mock;
  expect(calls[0][0]).toEqual(calls[1][0]);
  expect(stripAnsi(calls[0][0])).toMatchSnapshot();
});

test('display unsupported command info (run-ios)', () => {
  run(['run-ios']);
  const { calls } = console.log.mock;
  const message = calls[1].join(' ');
  expect(stripAnsi(message)).toMatchSnapshot();
});

test('display command help (start)', () => {
  run(['start', '--help']);
  const { calls } = console.log.mock;
  expect(stripAnsi(calls[0][0])).toMatchSnapshot();
});

test('run a command without options (init)', async () => {
  run(['init']);
  await flushPromises();
  expect(init.action).toBeCalledWith({});
});
