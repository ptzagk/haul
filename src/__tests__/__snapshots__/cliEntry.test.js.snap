// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display command help (start) 1`] = `
"haul start [options]
Starts a new webpack server

Options:

  --port [number]                 Port to run your webpack server, 8081 by default
  --dev [true|false]              Whether to build in development mode, true by default
  --minify [true|false]           Whether to minify the bundle, 'true' by default when dev=false
  --platform <ios|android|all>    Platform to bundle for
  --config [string]               Path to config file, eg. webpack.haul.js, webpack.haul.js by
                                  default"
`;

exports[`display help 1`] = `
"
Usage: haul [command] [options]

Options:

  --version  print the version number
  --help     print usage information

Commands:

  init       Generates necessary configuration files
  start      Starts a new webpack server
  bundle     Builds the app bundle for packaging

Run haul COMMAND --help for more information on specific commands
"
`;

exports[`display unsupported command info (run-ios) 1`] = `
" INFO  This command is not yet supported

Call react-native run-ios instead"
`;
